build_tools/upload_logs_to_s3.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+"""
+upload_logs_to_s3.py
+
+Uploads log files and index.html to an S3 bucket using the AWS CLI.
+"""
+
+import os
+import sys
+import glob
+import shutil
+import argparse
+import subprocess
+from pathlib import Path
+
+
+def log(*args, **kwargs):
+    print(*args, **kwargs)
+    sys.stdout.flush()
+
+
+def check_aws_cli_available():
+    if shutil.which("aws") is None:
+        log("[ERROR] AWS CLI not found in PATH.")
+        sys.exit(1)
+
+
+def run_aws_cp(src: str, dest: str, content_type: str = None):
+    cmd = (
+        ["aws", "s3", "cp", src, dest, "--recursive"]
+        if os.path.isdir(src)
+        else ["aws", "s3", "cp", src, dest]
+    )
+    if content_type:
+        cmd += ["--content-type", content_type]
+    try:
+        log(f"[INFO] Running: {' '.join(cmd)}")
+        subprocess.run(cmd, check=True)
+    except subprocess.CalledProcessError as e:
+        log(f"[ERROR] Failed to upload {src} to {dest}: {e}")
+
+
+def upload_logs_to_s3(
+    bucket_name: str, run_id: str, amdgpu_family: str, build_dir: Path
+):
+    log_dir = build_dir / "logs"
+    s3_base_path = f"s3://{bucket_name}/{run_id}-linux/logs/{amdgpu_family}"
```

Comment:
We need handling in this script for both `-linux` and `-windows`. Can either infer that from the logs themselves, the sys/platform, or some explicit argument.

My intuition would be to have the workflow do this:

```bash
python build_tools/upload_logs_to_s3.py \
  --build-dir=${BUILD_DIR} \
  --s3-base-path="s3://therock-artifacts/${{github.run_id}}-linux/logs/${{env.AMDGPU_FAMILIES}}/"
```

Then the script would not need to know anything about amdgpu_family or run_id. Less plumbing that way, and the scripts would be runnable locally without needing to think about github actions. The disadvantage to that approach though is that the path structure would be only determined in a workflow file... maybe that could be its own `get_s3_base_path_for_workflow.py` script?

Reply:
```
python build_tools/upload_logs_to_s3.py \
  --build-dir=${BUILD_DIR} \
  --s3-base-path="s3://therock-artifacts/${{github.run_id}}-linux/logs/${{env.AMDGPU_FAMILIES}}/"
```
This will reduce plumbing a lot. I will implement to see how it goes.

Reply:
Addressed